integration-tests/integration.test.js

```diff
@@ -282,4 +282,18 @@ describe('Tram-One', () => {
 		const keys = Object.keys(global.tramSpace)
 		expect(keys.length).toBeGreaterThan(0)
 	})
+
+	it('should not trigger side-effects of raw objects', () => {
+		// mount the app on the container
+		const container = document.createElement('div')
+		start(container, frozenCounter)
+
+		// verify that the button is rendered with the default text
+		expect(getByRole(container, 'button')).toHaveTextContent('clicks: 0')
+		// click on the click button to trigger update
+		fireEvent.click(getByRole(container, 'button'))
+
+		// verify that the button was not updated
+		expect(getByRole(container, 'button')).toHaveTextContent('clicks: 0')
+	})
```

Comment:
This test verifies that a raw (non-proxied) object will not cause side effects when it is mutated.